src/track/beatfactory.cpp

```diff
@@ -79,15 +79,19 @@ mixxx::BeatsPointer BeatFactory::makePreferredBeats(
         const mixxx::Bpm constBPM = BeatUtils::makeConstBpm(
                 constantRegions, sampleRate, &firstBeat);
         firstBeat = BeatUtils::adjustPhase(firstBeat, constBPM, sampleRate, beats);
-        auto pGrid = mixxx::BeatGrid::makeBeatGrid(
-                sampleRate, constBPM, firstBeat.toNearestFrameBoundary(), subVersion);
-        return pGrid;
+        if (firstBeat.isValid()) {
```

Comment:
The method is only called by analyzerbeats. Can we move the check there and leave the debug assertion intact?

Reply:
Sure. I only wonder why the assertion was triggered by firstBeat.toNearestFrameBoundary(), because adjustPhase() already requires a valid value? firstBeat is invalidated by makeConstBpm().

Reply:
Em, no, I don't think we can't move this check out of BeatFactory::makePreferredBeats(). Will test again.

Also edge cases with a single beat are not handled correctly. It doesn't make any sense to infer a tempo for less than 2 beats.